app/src/main/kotlin/com/wire/android/ui/common/textfield/StateSyncingModifier.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.ui.common.textfield
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.text.input.TextFieldCharSequence
+import androidx.compose.foundation.text.input.TextFieldState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.node.ModifierNodeElement
+import androidx.compose.ui.node.ObserverModifierNode
+import androidx.compose.ui.node.observeReads
+import androidx.compose.ui.platform.InspectorInfo
+import androidx.compose.ui.text.input.TextFieldValue
+import io.github.esentsov.PackagePrivate
+
+/**
+ * Enables us to temporarily still use TextFieldValue and onValueChanged callback instead of TextFieldState directly,
+ * also allows us to get selection updates as by default BasicTextField2 callback only gives a String without selection.
+ * @sample androidx.compose.foundation.samples.BasicTextFieldWithValueOnValueChangeSample
+ * TODO: Remove this class once all WireTextField usages are migrated to use TextFieldState.
+ */
+@PackagePrivate
+internal class StateSyncingModifier(
+    private val state: TextFieldState,
+    private val value: TextFieldValue,
+    private val onValueChanged: (TextFieldValue) -> Unit,
+) : ModifierNodeElement<StateSyncingModifierNode>() {
+
+    override fun create(): StateSyncingModifierNode = StateSyncingModifierNode(state, onValueChanged)
+
+    override fun update(node: StateSyncingModifierNode) {
+        node.update(value, onValueChanged)
+    }
+
+    @Suppress("EqualsAlwaysReturnsTrueOrFalse")
+    override fun equals(other: Any?): Boolean = false
```

Comment:
question: why is it needed?

Reply:
In sample from Google there's a comment explaining that:
`// Always call update, without comparing the text. Update can compare more efficiently.`